packages/core/src/command/rate-limiting/ddb-store.ts

```diff
@@ -0,0 +1,323 @@
+        key,
+      });
+    }
+    return {
+      totalHits: entry.data.totalHits,
+      resetTime: this.buildDate(entry.data.resetTime),
+    };
+  }
+
+  /**
+   * Method to decrement a client's hit counter.
+   *
+   * @param key {string} - The identifier for a client.
+   *
+   * @public
+   */
+  async decrement(key: string): Promise<void> {
+    const { log } = out(`DynamoStore decrement - key ${key}`);
+    const item = await this.ddbUtils.update({
+      partition: this.prefixCountKey(key),
+      expression: "ADD totalHits :inc",
+      expressionAttributesValues: {
+        ":inc": -1,
+      },
+    });
+    const entry = rateLimitCountSchema.safeParse(item.Attributes);
+    if (!entry.success) {
+      const error = entry.error;
+      const msg = "Error parsing DynamoStore count entry";
+      log(`${msg} - error: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          key,
+          context: "dynamo-store.decrement",
+          error,
+        },
+      });
+      throw new MetriportError(msg, error, { method: "decrement", key });
+    }
+  }
+
+  /**
+   * Method to reset a client's hit counter.
+   *
+   * @param key {string} - The identifier for a client.
+   *
+   * @public
+   */
+  async resetKey(key: string): Promise<void> {
+    const { log } = out(`DynamoStore resetKey - key ${key}`);
+    const item = await this.ddbUtils.update({
+      partition: this.prefixCountKey(key),
+      expression: "SET totalHits = :totalHits REMOVE resetTime",
+      expressionAttributesValues: {
+        ":totalHits": 0,
+      },
+    });
+    const entry = rateLimitCountSchema.safeParse(item.Attributes);
+    if (!entry.success) {
+      const error = entry.error;
+      const msg = "Error parsing DynamoStore count entry";
+      log(`${msg} - error: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          key,
+          context: "dynamo-store.resetKey",
+          error,
+        },
+      });
+      throw new MetriportError(msg, undefined, { method: "resetKey", key });
+    }
+  }
+
+  /**
+   * Method to fetch a client's limit. Introduced by Metriport.
+   *
+   * @param key {string} - The identifier for a client.
+   *
+   * @returns {number} - The limit of hits within the window.
+   *
+   * @public
+   */
+  async getLimit(key: string, defaultLimit: number): Promise<number> {
+    const { log } = out(`DynamoStore get - key ${key}`);
+    const item = await this.getOrCreateLimit(key, defaultLimit);
+    const entry = rateLimitLimitSchema.safeParse(item);
+    if (!entry.success) {
+      const error = entry.error;
+      const msg = "Error parsing DynamoStore limit entry";
+      log(`${msg} - error: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          key,
+          context: "dynamo-store.getLimit",
+          error,
+        },
+      });
+      throw new MetriportError(msg, error, { method: "getLimit", key, defaultLimit });
```

Comment:
Since this is a "500", should we just throw and have it be captured by the default error handler?

Reply:
This gets called within `limit: async ()....` to could wrap there, but I think it's better here.

Unless I'm misunderstanding something...?

Reply:
Gotcha. I was thinking how to avoid sending 1,000's of events to Sentry in case this breaks.

Good as is for now, we need to revisit error handling anyways - https://metriport.slack.com/archives/C04DMKE9DME/p1731616511906149